### PR TITLE
[13.0][ADD] sale_order_product_availability_inline: new module

### PR DESCRIPTION
**Pull request by ernestotejeda**
Show product availability (free_qty) in sales order lines product drop-down.

Screenshot 2020-09-22 18:04:40

cc @Tecnativa TT25486

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please remove `Free to use: ` word there, and I'm debating about putting before or after the quantity between parenthesis (as there are very long product names).

**Comment by ernestotejeda** [comment]
@pedrobaeza , `Free to use:` removed.


> I'm debating about putting before or after the quantity between parenthesis (as there are very long product names).

Maybe before is better but I'm not sure so let me know if you think it's better.

**Review by rafaelbn — APPROVED** [review]
Functional test un runbot

File: sale_order_product_availability_inline/views/sale_views.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='order_line']/tree//field[@name='product_id']"
+                position="attributes"
+            >
+                <attribute name="context">{
+                        'so_product_stock_inline': True,
```
Comment (youring) [inline sale_order_product_availability_inline/views/sale_views.xml]: IMHO, is it better to using python_dict(depends on base_view_inheritance_extension) to append context, other than overriding? This way keeps other modules' context working.

Comment (pedrobaeza) [reply sale_order_product_availability_inline/views/sale_views.xml]: Can you please do a PR against our branch for adding this?

**Review by victoralmau — APPROVED** [review]

**Review by HaraldPanten — CHANGES_REQUESTED** [review]
Units availability are shown in the quotation report. IMO they shouldn't appear there, as it's an internal information and not an information for customers.

Steps:

1- Create a quotation for a random partner.
2- Add several lines with several stockable products. These products need to have several "free to use" stock units.
3- Save and print the quotation.

**This is the result:**

Captura de pantalla 2020-09-22 a las 9 57 28

**Comment by pedrobaeza** [comment]
The problem is not printing the report, but on onchange product that keeps the context and thus shows display_name with quantities. @ernestotejeda please overwrite the onchange and reset that context key before calling to super.

**Comment by ernestotejeda** [comment]
@pedrobaeza , @HaraldPanten  Changes done

File: sale_order_product_availability_inline/views/sale_views.xml (hunk 1 above)
Comment (youring) [reply sale_order_product_availability_inline/views/sale_views.xml]: Done. Just did my first PR to OCA, I'm not a developer, correct me if wrong.

Comment (pedrobaeza) [reply sale_order_product_availability_inline/views/sale_views.xml]: I have just merged it. You can check if it works soon on runbot.

**Comment by youring** [comment]
One not funcional related question here: why the free_qty field shows one decimal place, not respecting the digits (3) of "Product Unit of Measure"?

**Comment by pedrobaeza** [comment]
Because the quantity is shown as delivered in Python. Expect that if you have 3 decimals, they will be shown as well.

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍

**Comment by youring** [comment]
> 
> 
> Because the quantity is shown as delivered in Python. Expect that if you have 3 decimals, they will be shown as well.

Then how to show as integer while digit for "Product Unit of Measure" is 0?

**Comment by pedrobaeza** [comment]
You need to investigate that, as we don't have more budget for this.

**Pull request merged**